### DISCUSSION TITLE
What does "template_size" and "dont_expand" in the complete_settings mean?

### DISCUSSION THREAD

**Discussion by pwn2ooown** [Jatana/FastOlympicCoding#61] [Q&A] [answered]
I'm setting the completion of this incredible tool now but I have trouble understanding these two statements, can someboby explain the meaning of them? Thanks a lot.

**Comment by Jatana** [comment, marked as answer]
`template_size` is the number of template arguments a specific class requires. For example, `pair` requires two template arguments, `vector` requires one template argument, and `int` requires 0 template arguments.

If, for some reason, you do not want an alias to be expanded, you can specify it in the `dont_expand` option. For instance, normally, `vpii` is expanded into `vector<pair<int, int>>`. However, if you add `pii` to the `dont_expand` list, then `vpii` will be expanded only to `vector<pii>`.

**Reply by pwn2ooown** [reply]
Thank you Mr. Jatana for explaining these words! And really thank you for developing this powerful tool for CP!
